### DISCUSSION TITLE
micropython-SYN6988 : MicroPython library for the YuTone VoiceTX SYN6988 text to speech module

### DISCUSSION THREAD

**Discussion by scruss** [micropython/micropython#11891] [Show and tell]
Demo:

https://github.com/scruss/micropython-SYN6988/assets/425706/bf107dc3-3db7-4eb7-90a4-6cfe5986529f

Which is read from this:

```
[g2]Hello! [g1]你好![g2]
I can speak in English, and also in Chinese:
[g1]千里之行，始于足下。[g2]
which means: a journey of five hundred kilometres starts with the first step
[s1]I can speak slowly, [s9] and I can speak very fast[s5]
[v5]I can be [v10]loud or [v1]very quiet. [v5]
I can use [t1]low tones [t5]or [t8]high tones[t5]
[d][g0]my work here is done [x1]soundy[d]
```

using this: [scruss/micropython-SYN6988: MicroPython library for the VoiceTX SYN6988 text to speech module](https://github.com/scruss/micropython-SYN6988/tree/main)

The SYN6988 text to speech module is incredible value for money and works far better than I could have expected.

**Comment by mcauser** [comment]
```bash
$ say -v ralph thats awesome
```

**Reply by scruss** [reply]
if you like speech, you'll love
https://github.com/dectalk/dectalk
especially
https://webspeak.terminal.ink/

**Comment by mcauser** [comment]
I couldn't find any `YS-V6E` or `SYN6988` boards on AliExpress. Don't suppose you have a link?

I did find a smaller `SYN6288` board: https://www.aliexpress.com/item/1005003579005473.html
And a board labelled `SYN6988` but says SYN6658 on the silkscreen: https://www.aliexpress.com/item/1005004262579755.html

This looks like it: https://www.taobao.com/list/item/639119499765.htm

**Reply by scruss** [reply]
Yes, it's kind of messy. I have a SYN6288 and a XFS5152 board on the way. What I bought was this (now unavailable)
https://www.aliexpress.com/item/1005001580006007.html
which was sold as a XFS5152 and yet it has a SYN6988 on board

From what I understand (limited) the SYN6288, SYN6658 and SYN6988 are upwardly compatible. AliExpress often has chip silkscreens blurred. Your Taobao board link looks like mine. This looks like a SYN6988 despite the board stencil say SYN6658:
https://www.aliexpress.com/item/1005005501815195.html

(maybe I need to get a SYN6658 board now too ...)

**Reply by scruss** [reply]
Whoah there! I misspoke: sorry, @mcauser . The only boards that explicitly speak English are the SYN6988 and the SYN8089. The others are Chinese only. They may be able to make a few useful utterances, but they don't have the easy "type text, get English" feature of the SYN6988.

**Reply by scruss** [reply]
This seems to be a source: [High-end Speech Synthesis Module Chinese/English Speech Synthesis XFS5152 Real Pronunciation TTS - AliExpress](https://www.aliexpress.com/item/1005005348524077.html)
Despite the description, it's a YS-V6E with a SYN6988 silkscreen on the back and a SYN6988 fitted
